obelisk/cpp/obelisk_cpp/include/obelisk_node.h

```diff
@@ -0,0 +1,451 @@
+#pragma once
+#include <any>
+#include <chrono>
+#include <tuple>
+
+#include "obelisk_control_msgs/msg/position_setpoint.hpp"
+#include "obelisk_estimator_msgs/msg/estimated_state.hpp"
+#include "obelisk_sensor_msgs/msg/joint_encoder.hpp"
+#include "obelisk_sensor_msgs/msg/joint_encoders.hpp"
+#include "obelisk_sensor_msgs/msg/true_sim_state.hpp"
+#include "rcl_interfaces/msg/parameter_event.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+
+/**
+ * ObeliskNode class.
+ * A lifecycle node whose publishers and subscribers can only publish and
+ * subscribe to Obelisk messages.
+ */
+
+namespace obelisk {
+class ObeliskNode : public rclcpp_lifecycle::LifecycleNode {
+  public:
+    explicit ObeliskNode(
+        const std::string& node_name,
+        const rclcpp::NodeOptions& options  = rclcpp::NodeOptions(),
+        bool enable_communication_interface = true)
+        : LifecycleNode(node_name, options, enable_communication_interface),
+          CB_GROUP_NONE("None"),
+          CB_GROUP_MUTUALLY_EXEC("MutuallyExclusiveCallbackGroup"),
+          CB_GROUP_REENTRANT("ReentrantCallbackGroup") {
+        this->declare_parameter<std::string>("callback_group_config_strs", "");
```

Comment:
another reminder to declare the type explicitly here instead of supplying a bad default value. i'll stop commenting on this.